packages/stack/deployment/test/deployment.spec.js

```diff
@@ -0,0 +1,186 @@
+import Deployment from '../src/';
+
+const events = {
+  listeners: {},
+  setCommandHandler: sinon.spy((cmd, fn) => {
+    events.listeners[cmd] = fn;
+  }),
+  request: sinon.spy((cmd, ...args) => {
+    assert(events.listeners[cmd]);
+    events.listeners[cmd](...args);
+  }),
+  emit: (name) => {}
+};
+
+const plugins = {
+  listeners: {},
+  _plugin: {
+    listeners: {},
+    registerActionForEvent: (name, cb) => {
+      if (!plugins._plugin.listeners[name]) {
+        plugins._plugin.listeners[name] = [];
+      }
+      plugins._plugin.listeners[name].push(cb);
+    }
+  },
+  createPlugin: (name, opts) => {
+    return _plugin;
+  },
+  emitAndRunActionsForEvent: sinon.spy((name, options, cb) => {
+    if (plugins._plugin.listeners[name]) {
+      plugins._plugin.listeners[name].forEach(fn => {
+        fn(options, cb);
+      });
+    }
+  })
+};
+
+const fakeEmbark = {
+  events,
+  logger: {
+    info: () => {}
+  },
+  config: {
+    blockchainConfig: {}
+  },
+  assert: {
+    hasCommandHandler: (cmd) => {
+      sinon.assert.calledWith(fakeEmbark.events.setCommandHandler, cmd);
+    },
+    hasRequested: (cmd) => {
+      sinon.assert.calledWith(fakeEmbark.events.request, cmd);
+    },
+    hasRequestedWith: (cmd, ...args) => {
+      sinon.assert.calledWith(fakeEmbark.events.request, cmd, ...args);
+    },
+  },
+  teardown: () => {
+    fakeEmbark.events.listeners = {};
+    plugins._plugin.listeners = {};
+  }
+};
+
+describe('stack/deployment', () => {
+
+  let deployment;
+  let deployedContracts = [];
+
+  let beforeAllAction, beforeDeployAction, shouldDeployAction, deployedAction, afterAllAction;
+  let deployFn;
+  let doneCb;
+
+  beforeEach(() => {
+    deployment = new Deployment(fakeEmbark, { plugins });
+
+    beforeAllAction = sinon.spy((params, cb) => { cb(null, params); });
+    beforeDeployAction = sinon.spy((params, cb) => { cb(null, params); });
+    shouldDeployAction = sinon.spy((params, cb) => { cb(null, params); });
+    deployedAction = sinon.spy((params, cb) => { cb(null, params); });
+    afterAllAction = sinon.spy((params, cb) => { cb(null, params); });
+
+    deployFn = sinon.spy((contract, done) => {
+      deployedContracts.push(contract);
+      done(null, {}); // deployer needs to finish with a receipt object
+    });
+
+    doneCb = sinon.fake();
+  });
+
+  afterEach(() => {
+    deployedContracts = [];
+    fakeEmbark.teardown();
+    sinon.restore();
+  });
+
+  test('it should register deployFn and deploy contract by using it', () => {
+
+    let testContract = { className: 'TestContract', shouldDeploy: true };
+
+    plugins._plugin.registerActionForEvent('deployment:contract:beforeDeploy', beforeDeployAction);
+    plugins._plugin.registerActionForEvent('deployment:contract:shouldDeploy', shouldDeployAction);
+    plugins._plugin.registerActionForEvent('deployment:contract:deployed', deployedAction);
+
+    events.request('deployment:deployer:register', 'ethereum', deployFn);
```

Comment:
The blockchain type `ethereum` is hard-coded in the module at the moment. This is very likely going to be fragile and should be fixed soon.